src/internal/packager/images/pull.go

```diff
@@ -308,6 +311,27 @@ func (i *ImageConfig) PullImage(src string, spinner *message.Spinner) (img v1.Im
 			return nil, false, fmt.Errorf("failed to load image from docker daemon: %w", err)
 		}
 	} else {
+		refInfo, err := transform.ParseImageRef(src)
```

Comment:
Would really love comments on this logic here because it's a lot, but isn't necessary, and is looking a bit cleaner.